lib/api/src/modules/release-notes.ts

```diff
@@ -0,0 +1,94 @@
+import { RELEASE_NOTES_DATA } from 'global';
+import semver from '@storybook/semver';
+import memoize from 'memoizerific';
+
+import { ModuleFn, State } from '../index';
+
+export interface ReleaseNotes {
+  success?: boolean;
+  isAutoMessagingEnabled?: boolean;
+  currentVersion?: string;
+  history?: [string];
+}
+
+const getReleaseNotesData = memoize(1)(
+  (): ReleaseNotes => {
+    try {
+      return { ...(JSON.parse(RELEASE_NOTES_DATA) || {}) };
+    } catch (e) {
+      return {};
+    }
+  }
+);
+
+export interface SubAPI {
+  releaseNotesVersion: () => string;
+  showReleaseNotesOnLaunch: () => Promise<boolean>;
+  didViewReleaseNotes: () => boolean;
+  setDidViewReleaseNotes: () => Promise<void>;
+}
+
+export const init: ModuleFn = ({ store }) => {
+  let initState: Promise<State>;
+  const releaseNotesData = getReleaseNotesData();
+
+  const api: SubAPI = {
+    releaseNotesVersion: () => releaseNotesData.currentVersion,
+    showReleaseNotesOnLaunch: async () => {
+      await initState;
+
+      const {
+        releaseNotes: { showOnLaunch },
+      } = store.getState();
+
+      return showOnLaunch;
+    },
+    didViewReleaseNotes: () => {
+      const {
+        releaseNotes: { releaseNotesViewed },
+      } = store.getState();
+
+      return (
+        releaseNotesData.success && releaseNotesViewed.includes(releaseNotesData.currentVersion)
+      );
+    },
+    setDidViewReleaseNotes: async () => {
+      const { releaseNotes } = store.getState();
+      const uniqueVersionsViewed = Array.from(
+        new Set([
+          ...releaseNotes.releaseNotesViewed,
+          ...(releaseNotesData.success ? [releaseNotesData.currentVersion] : []),
+        ])
+      );
+
+      await store.setState(
+        { releaseNotesViewed: uniqueVersionsViewed },
+        { persistence: 'permanent' }
+      );
```

Comment:
I thought we were going to keep things simple and not track any persistent state in the browser. Did that change?

Reply:
I wanted to, but when it came down to implementation, I wasn't sure how to do that & prevent the release notes from re-opening every time you re-loaded your Storybook (a browser refresh). It seemed like I needed to track that on the client to prevent them from getting "redirected" to the release notes every time they refreshed the page. Do you have any ideas on how to prevent doing this?

Reply:
Ahh, got it. Yeah that makes sense. Thanks for the clarification